src/components/ModalWrapper.jsx

```diff
@@ -0,0 +1,24 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+
+class ModalWrapper extends PureComponent {
+  // TODO remove mock styling
+  render() {
+    const { children } = this.props;
```

Comment:
Add modalKey prop; connect ModalWrapper to redux and move here logic about hiding/showing current modal.